src/utils/UserReportAPILayer.tsx

```diff
@@ -0,0 +1,59 @@
+import { gql, request } from 'graphql-request'
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+
+import { LastDetection } from '../components/LastDetection'
+//Feel free to add additional GraphQL queries as needed
+
+interface UserReport {
+  date: number | string
+  setDate: Dispatch<SetStateAction<number | string>>
+}
+
+export const UserReportAPILayer = (props: UserReport): JSX.Element => {
+  const [date, setDate]: any = useState('')
+
+  useEffect(() => {
+    const detectionQuery: string = gql`
+      query getDetections($pagination: Pagination!) {
+        detections(pagination: $pagination) {
+          meta {
+            currentPage
+            previousPage
+            nextPage
+            totalEntries
+            totalPages
+          }
+          entries {
+            timestamp
+          }
+        }
+      }
+    `
+    const variables: { pagination: { page: number; pageSize: number } } = {
+      pagination: {
+        page: 1,
+        pageSize: 10,
+      },
+    }
+    request(
+      'https://live.orcasound.net/graphql',
+      detectionQuery,
+      variables
+    ).then((data: any): any => {
+      //Reformatting of time data for LastDetection component
+      //Array time data
```

Comment:
Can you give an example of the formation of "timestamp" and what you're converting it into so that the code is easier to understand for readability and correctness?